repository/fcm_token_repository.py

```diff
@@ -55,3 +57,66 @@ def unregister_token(self, user_id: int, fcm_token: str) -> None:
                 logging.error(f"Database error while unregistering FCM token for user {user_id}:{e}")
                 session.rollback()
                 raise e
+
+    def get_fcm_token(self, user_id: int) -> List[str]:
+        """
+        Get FCM tokens for a user
+
+        :param user_id: The user id
+        :return: A list of FCM tokens for a given user.
+        """
+        with session_scope() as session:
+            try:
+                # Query to get all tokens for the given user_id
+                existing_tokens = session.query(FCMToken).filter_by(user_id=user_id).all()
+
+                if not existing_tokens:
+                    logging.info(f"No active FCM token found for user {user_id}")
+                    return []
+
+                fcm_token_list = [token.fcm_token for token in existing_tokens]
+                return fcm_token_list
+
+            except SQLAlchemyError as e:
+                logging.error(f"Database error while retrieving FCM token for user {user_id}: {e}")
+                raise e
+
+    def notify_user(
+            self,
+            user_id: Optional[int] = None,
```

Comment:
I think you should refine what is and isn't optional here. I think I understand why you've done this, but I would prefer if the arguments were in a weird order but correctly marked as optional/required. I'm also happy with you just doing a check in the body that `title` and `body` aren't `None`, like you've done with `user_id` and `fcm_token_list`.

Reply:
BTW I'm assuming title and body shouldn't be null because notification service expects them not to be, I could be wrong though so feel free to correct.